### ISSUE TITLE
Error 403 (Forbidden) 

### ISSUE THREAD

**Issue by mistermaps** [lfos/calcurse#174]
Hi,
Just set up calcurse-caldav and keep encountering a 403 Forbidden error. Read through the suggested threads. I've made sure I don't have any tasks, and the calendar syncs locally for a bit before encountering an error:
`calcurse-caldav --init=two-way --authcode [redacted]
/usr/bin/calcurse-caldav:501: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
  config.readfp(open(configfn))
Connecting to apidata.googleusercontent.com...
Importing new object 63684163161.
Importing new object 63674166891.
Importing new object 63583790394.
Importing new object 63608393613.
Importing new object 63583817627.
Importing new object 63659965984.
Importing new object 63684162943.
Importing new object 63670217978.
Importing new object 63682213212.
Importing new object 63579851580.
Importing new object 63669258179.
Importing new object 63583758247.
Importing new object 63684162975.
Importing new object 63608393537.
Importing new object 63673695717.
Importing new object 63665653032.
Importing new object 63684131458.
Importing new object 63617720824.
Importing new object 63674733652.
Importing new object 63665036268.
Importing new object 63683190913.
Importing new object 63664057456.
Importing new object 63684141912.
Importing new object 63665653065.
Importing new object 63669346844.
Importing new object 63654414197.
Importing new object 63684163160.
Importing new object 63681311898.
Importing new object 63684163230.
Importing new object 63674810800.
Importing new object 63678625258.
Importing new object 63596720758.
Importing new object 63680263280.
Importing new object 63655974492.
Importing new object 63684163097.
Importing new object 63684163231.
Pushing new object db25adac04a3bb2641c96a79a0170e6cebca9428 to the server.
Pushing new object 06ba68f081ed1816e0356e4894653943aad4e97c to the server.
Pushing new object 4b542111010ce5806bb4e52d395e78ac7287cfa3 to the server.
Pushing new object b1ac742f076c1b4d84742c26f332e6275e1739b9 to the server.
Pushing new object aa5c26e1ac4023abc11db2bc6c08d22457642d3a to the server.
Pushing new object d4672531a839004b83217fdf43080e4cc3017a06 to the server.
Pushing new object 9ad35f28de461ef6be44ceb1cf5727b71c7062b7 to the server.
error: The server at apidata.googleusercontent.com replied with HTTP status code
error: 403 (Forbidden) while trying to access https://apidata.googleusercontent.
error: com/caldav/v2/[redacted]/events/9ad35f28de461ef6be44ceb1cf5727b71
error: c7062b7.ics.`

This, as displayed here, adds content to my local apts file, but doesn't appear to add anything to the server as far as I can see.

I've made sure both calcurse and calcurse-dev are at the latest build, which hasn't rectified the issue. I'm assuming there's got to be an error somewhere in my config file - if this is correct, let me know what info you'd like me to post and I'll share it here with any sensitive information redacted.

Thanks!

**Comment by mistermaps** [comment]
Updated:

It looks like it's been pushing to my calendar, but I think a very odd error is happening here.

On my initial init I had made a mistake and had tasks on my todo list locally. After reading the errors here, I deleted them.

However, I believe they've somehow been pushed to the server - which I'm led to believe is impossible - because after blanking out the file (and confirming it's blank), I attempt a sync and see that all of the tasks that were previously there have been returned. I've made sure that todo.new is also blanked out.

It's also making redundancies whenever synced, both locally and remotely - we can fix this aspect later.

**Comment by mistermaps** [comment]
I managed to solve this in a way by changing from two way to keep-local and cutting my losses with regards to erasing the old history of this calendar (there was nothing I needed to keep there, thankfully.)

... but now I'm getting an error that I'm not sure how to rectify, seemingly apropos of nothing. I didn't make any changes to the config.

`/usr/bin/calcurse-caldav:501: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
  config.readfp(open(configfn))
Connecting to apidata.googleusercontent.com...
Traceback (most recent call last):
  File "/usr/bin/calcurse-caldav", line 604, in <module>
    cred = run_auth(authcode)
  File "/usr/bin/calcurse-caldav", line 157, in run_auth
    credentials = init_auth(client_id, client_secret, scope, redirect_uri, authcode)
  File "/usr/bin/calcurse-caldav", line 127, in init_auth
    credentials = oauth2_client.step2_exchange(authcode)
  File "/home/maps/.local/lib/python3.7/site-packages/oauth2client/_helpers.py", line 133, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/maps/.local/lib/python3.7/site-packages/oauth2client/client.py", line 2089, in step2_exchange
    raise FlowExchangeError(error_msg)
oauth2client.client.FlowExchangeError: invalid_grantBad Request
`
